bindings/python/iio.py

```diff
@@ -1081,6 +1146,54 @@ def __next__(self):
         return Block(self._buffer, self._block_size, next_hdl)
 
 
+class EventStream(_IIO_Object):
+    def __init__(self):
+        # Prevent the EventStream from being initialized manually.
+        raise NotImplementedError
+
+    @classmethod
+    def _create(cls, device, stream):
+        self = cls.__new__(cls)
+        super(EventStream, self).__init__(stream, device)
+
+        return self
+
+    def read(self, nonblock = True):
+        """
+        Read one event from the stream.
+
+        :param nonblock: type=bool
+            If set, return None if there are no events in the stream.
+            If unset, it will wait until an event arrives.
+
+        returns: type=(iio.Event, (iio.Channel, iio.Channel))
+            The first element of the tuple is the Event read.
+            The second element is a tuple itself, that contains the
+            two channels related to the event. For non-differential
+            events the second channel will always be None.
+        """
+        event = Event()
+
+        try:
+            _evstream_read(self._hdl, _byref(event), nonblock)
+        except OSError as err:
+            if err.errno == 11: # EAGAIN
+                return None
+            else:
+                raise
+
+        chn1 = None
+        chn2 = None
+
+        try:
+            dev = self._parent
+            chn1 = Channel(dev, _ev_get_channel(_byref(event), dev._device, False))
+            chn2 = Channel(dev, _ev_get_channel(_byref(event), dev._device, True))
+        except Exception:
+            pass
```

Comment:
Need to be more explicit for the exception

Reply:
I changed it to `OSError`.